Projects/Server/Utilities/SplitMix64.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Server
+{
+  public class SplitMix64
+  {
+    private ulong x;
+
+    public SplitMix64(ulong seed) => x = seed;
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public ulong Next() {
+      ulong z = x += 0x9e3779b97f4a7c15;
+      z = (z ^ (z >> 30)) * 0xbf58476d1ce4e5b9;
+      z = (z ^ (z >> 27)) * 0x94d049bb133111eb;
+      return z ^ (z >> 31);
+    }
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public void FillArray(ulong[] arr)
+    {
+      if (arr == null)
+        return;
+
+      for(int i = 0; i < arr.Length; i++) arr[i] = Next();

```

Comment:
I don't know what the style guides are here strictly, but the rest of the code base puts a space after the for and always puts the body on a separate line.